File: Stack/PrefixtoPostfix.cpp
#include <iostream>
#include <stack>
#include <string>

using namespace std;

string preToPost(string s) {
    // Write Your Code Here
    int n = s.length();
    string res = "";
    stack<string> st;
    for(int i = n - 1; i >= 0; i--){
        if(isalpha(s[i]) || isdigit(s[i])){
            string temp(1,s[i]);
            st.push(temp);
        }else{
            string op1 = st.top();
            st.pop();
            string op2 = st.top();
            st.pop();
            string temp = op1 + op2 + s[i];
            st.push(temp);
        }
    }
    res = st.top();
    return res;






JAVA - 


class Solution {
    static String preToPost(String pre_exp) {
        // code here
                Stack<String> st = new Stack<>();
        int n = pre_exp.length();

        // Traverse the prefix expression from right to left
        for (int i = n - 1; i >= 0; i--) {
            char ch = pre_exp.charAt(i);

            if (Character.isLetterOrDigit(ch)) {
                // Push operands (numbers or variables) onto the stack
                st.push(String.valueOf(ch));
            } else {
                // Pop two operands from the stack
                String op1 = st.pop();
                String op2 = st.pop();

                // Create the postfix expression and push it back to the stack
                String expr = op1 + op2 + ch;
                st.push(expr);
            }
        }

        // The final element of the stack is the postfix expression
        return st.pop();
    }
}

File: Stack/Infix,Prefix,Postfix/PrefixtoPostfix.java
class Solution {
    // Function to convert a prefix expression to a postfix expression.
    static String preToPost(String pre_exp) {
        int n = pre_exp.length(); // Get the length of the prefix expression
        Stack<String> st = new Stack<>(); // Initialize a stack to hold intermediate results
        
        // Loop through each character in the prefix expression in reverse order
        for (int i = n - 1; i >= 0; i--) {
            char ch = pre_exp.charAt(i); // Get the current character
            
            // If the character is a letter, push it onto the stack as a string
            if (Character.isLetter(ch)) {
                st.push(String.valueOf(ch));
            } 
            // If the character is an operator
            else {
                // Pop two elements from the stack
                String c1 = st.pop();
                String c2 = st.pop();
                
                // Form a new string with the format (operand1 operand2 operator)
                String temp = c1 + c2 + ch;
                
                // Push the new string back onto the stack
                st.push(temp);
            }
        }
        // The final result is the only element left in the stack
        return st.pop();
    }
}
